Week3/homework/step2-6.js

```diff
@@ -4,13 +4,24 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = arr.reduce((total, item) => total.concat(item));
```

Comment:
Since your `flattenArray3d` function can flatten any array, you might have simply called it from within your `flattenArray2d`:

```js
function flattenArray2d(arr) {
  return flattenArray3d(arr);
}
```

This will work fine, but ESLint will complain. It expects functions to be defined before they are used, even though JavaScript hoists function declaration. To shut up ESLint, just move the `flattenArray2d` declaration below the `flattenArray3d` declaration.